examples/transpose/transpose_dist_object.cpp

```diff
@@ -0,0 +1,380 @@
+// Copyright (c) 2019 Weile Wei
+// Copyright (c) 2019 Maxwell Reeser
+// Copyright (c) 2019 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+///////////////////////////////////////////////////////////////////////////
+/// This is an example of distributed matrix transposition using dist_object.
+///
+/// A distributed object is a single logical object partitioned across
+/// a set of localities. (A locality is a single node in a cluster or a
+/// NUMA domian in a SMP machine.) Each locality constructs an instance of
+/// dist_object<T>, where a value of type T represents the value of this
+/// this locality's instance value. Once dist_object<T> is conctructed, it
```

Comment:
`concructed`

Reply:
I also modified some descriptions for constructors and member functions. Please review.
https://github.com/STEllAR-GROUP/hpx/blob/dist_object_pr/hpx/lcos/distributed_object.hpp#L301